src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java

```diff
@@ -294,7 +311,9 @@ public void checkCoupling() {
          */
         private boolean isSignificant(String candidateClassName) {
             return !excludedClasses.contains(candidateClassName)
-                    && !candidateClassName.startsWith("java.lang.");
+                    && !candidateClassName.startsWith("java.lang.")
+                    && (excludeClassesRegexp.isEmpty()
+                    || !excludeClassesRegexpPattern.matcher(candidateClassName).find());
```

Comment:
Is is not correct. ```find()``` attempts to find the next subsequence of the input sequence that matches the pattern. You need to use matches(). Check doc: https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html .